Improve operation of ImagePicking

tgreve15 opened the pull request:
Reduced the number of clicks required to decide on a new image for a new or existing pot by moving the decision to the ImageList directly. Also included a few additional black and white images just in case they are desired.

seveneightn9ne reviewed (changes requested):
Overall looks good! Very nice to have fewer clicks. I'd like to request couple changes though:

- For visual consistency, let's switch back to using the material icon font instead of static image icons. You can search through the options [here](https://material.io/resources/icons/?style=baseline). I think [add_a_photo](https://material.io/resources/icons/?search=add_a_photo&style=baseline) and [add_photo_alternate](https://material.io/resources/icons/?search=add_photo_alternate&style=baseline) would be good. Funny their "+" buttons are on opposite sides though.
- I'd like to tweak the visual style of the buttons a bit. Since the app is styled following Material Design, I usually look for inspiration at material.io. Could you try to emulate the [toggle button](https://material.io/components/buttons#toggle-button) appearance? Specifically:
  - White background (for dark mode, I think the dark grey you have already looks good)
  - Light grey border
  - Black icon color
  - Rounded outside borders
  - the 2 buttons are touching and their touching borders aren't rounded
  - Uniform margin around the outsides of the buttons (right now you have what looks like less margin on the right edge)
I think the rectangular shape & position of them is fine as you have it. Ideally it would be nice to make the buttons square but I don't see a way to make that work visually since they have a lot of vertical space to take up.

tgreve15 commented:
Ahhh Material Icons - that makes sense. I didn't understand what was going
on previously, and couldn't get sizing working, which is why I switched to
images. I tried the two you suggested, but found the one without the
camera, the + would show far to the left of the icon on a new item, or just
be shown without the rest of the image on smaller, so for the moment I went
to two without the +. I will check the alternates again before submitting
changes to see what looks best.

I will still play around with the formatting of this, but for the moment
wanted to send you a few I have been playing with. I know you were talking
about one with borders etc, but one I accidentally came across I think
could be better. Some of the attached have the two buttons at the top left
which means they are equally visible when opening the view and clear to
press. This is also the case on the landscape view where using the current
approach would have the bottom button off screen until scrolling down.
I have also included in the attached both dark and light themes, with icons
in black/white and two shades of gray. Personally the black and white look
very stark to me (perhaps related to the background of my image as well),
although the lighter gray on black and darker gray on white seemed to fit
in a bit better I thought.

The two of the attached I like best are 215912 and 215939 (also on the two
landscape ones 220232 and 220213) for positioning and color, but let me
know your preference and I will finalize using those making sure I have
your other suggestions in place. Alternatively I can commit my changes with
an option to easily switch between them and let you play to see what looks
best for you.

btw considering your suggestions, I can stop with what I was working on,
since it is no longer needed.. :) A component (and asset folder structure)
that provided a cleaner way of displaying icons correctly, switching
between dark/light themes automatically. Either way, learning lots.

Thanks
Todd

On Fri, 12 Mar 2021 at 14:08, Jess Kenney ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> Overall looks good! Very nice to have fewer clicks. I'd like to request
> couple changes though:
>
>    - For visual consistency, let's switch back to using the material icon
>    font instead of static image icons. You can search through the options
>    here <https://material.io/resources/icons/?style=baseline>. I think
>    add_a_photo
>    <https://material.io/resources/icons/?search=add_a_photo&style=baseline>
>    and add_photo_alternate
>    <https://material.io/resources/icons/?search=add_photo_alternate&style=baseline>
>    would be good. Funny their "+" buttons are on opposite sides though.
>    - I'd like to tweak the visual style of the buttons a bit. Since the
>    app is styled following Material Design, I usually look for inspiration at
>    material.io. Could you try to emulate the toggle button
>    <https://material.io/components/buttons#toggle-button> appearance?
>    Specifically:
>       - White background (for dark mode, I think the dark grey you have
>       already looks good)
>       - Light grey border
>       - Black icon color
>       - Rounded outside borders
>       - the 2 buttons are touching and their touching borders aren't
>       rounded
>       - Uniform margin around the outsides of the buttons (right now you
>       have what looks like less margin on the right edge)
>       I think the rectangular shape & position of them is fine as you
>       have it. Ideally it would be nice to make the buttons square but I don't
>       see a way to make that work visually since they have a lot of vertical
>       space to take up.
>
>

seveneightn9ne commented:
Yeah, I like the ones you suggested. I would add a bit more vertical spacing between them to make sure their click boxes are nice and big. I wonder whether a thin border would be better to make them look more clearly like clickable buttons. For some reason to me they don't look obviously clickable. what do you think?

A component to make dealing with icons easier does sound nice, if you have something good there feel free to submit a separate PR for that, it might come in handy!

tgreve15 commented:
Finally submitting material icon updates I think. Let me know if my commit's from today didn't make it through. 
FYI I have another change being submitted switching the coffee mug icon to material icons as well and it seems there may be a few style changes for it in this, and the coffee icons removed. I couldn't see how to prevent the icon removals from the submission.

In ImagePicker I suspect the one currently visible is preferred, but feel free to un-comment the other ones to try them to see if you prefer their look.

seveneightn9ne commented:
Looking good! I tried out a small style tweak that I like, what do you think?

<img width="398" alt="Screen Shot 2021-03-26 at 10 29 29 PM" src="https://user-images.githubusercontent.com/751065/112707591-f60ef480-8e82-11eb-8c11-348749891b94.png">

I also added a placeholder image when there's no other image, because the whitespace looked weird to me:

<img width="399" alt="Screen Shot 2021-03-26 at 10 29 41 PM" src="https://user-images.githubusercontent.com/751065/112707594-f9a27b80-8e82-11eb-90ee-a4f7d32a96f8.png">

You can apply my [changes](https://github.com/seveneightn9ne/pottery-log/files/6215009/changes.txt) with `git apply` and add them on to this PR (I'm not sure if there's a better way to add my own changes?)

Otherwise, I think it's pretty much ready to go! Just make sure to remove all the extra commented out stuff.

tgreve15 commented:
Looks good to me.

Not sure I can tell you an easier way to share changes, it took me far too
long to realize you had a link to a file with your changes that I should
download and reference. Wondered why "git apply" never completed (without
the filename passed). Finally worked that out and the changes didn't apply
(had made minor formatting change to styles perhaps) so I manually copied
them over.

So I have now applied your changes and cleaned up style and other files
removing unnecessary parts of my work and committed all. Please accept and
apply them when you can.

A few comments
* In dark mode the background of the ElevatedView showed white so I fixed
that with a new style element.
* For a proper image on light mode you can see a bit of a line below the
image picker due to the elevation I gather. Don't see it any other time. If
you decide against using the ElevatedView to get rid of that, please commit
my changes and update it yourself unless there is a reason not to do it
that way.
* For some reason, last night it was showing in light mode even though I
have the theme set for default and it was night time. I checked a few times
today and it seemed to be working ok, so not sure what happened there. I
did run a yarn update this evening, but wouldn't think that would make a
difference there.

Is there a way to test the automatic change of the material design theme,
or is that manually changing the time on your device the only way?

Is it possible to enable the wiki on github and give me access to
contribute to that? I was thinking of putting a few articles there - one
about what I learnt about the potential ShowImage control. Not finished and
perhaps never needed, but findings perhaps useful in future.

Potential New Issue
* ImagePage always shows black background, regardless of theme, suspect for
completeness this shouldn't be the case. What is your thought?

Hopefully now we are done with this change :)

Thanks
Todd



On Sat, 27 Mar 2021 at 13:41, Jess Kenney ***@***.***> wrote:

> Looking good! I tried out a small style tweak that I like, what do you
> think?
>
> [image: Screen Shot 2021-03-26 at 10 29 29 PM]
> <https://user-images.githubusercontent.com/751065/112707591-f60ef480-8e82-11eb-8c11-348749891b94.png>
>
> I also added a placeholder image when there's no other image, because the
> whitespace looked weird to me:
>
> [image: Screen Shot 2021-03-26 at 10 29 41 PM]
> <https://user-images.githubusercontent.com/751065/112707594-f9a27b80-8e82-11eb-90ee-a4f7d32a96f8.png>
>
> You can apply my changes
> <https://github.com/seveneightn9ne/pottery-log/files/6215009/changes.txt>
> with git apply and add them on to this PR (I'm not sure if there's a
> better way to add my own changes?)
>
> Otherwise, I think it's pretty much ready to go! Just make sure to remove
> all the extra commented out stuff.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/seveneightn9ne/pottery-log/pull/19#issuecomment-808629377>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADZS5MVXSJIIJXXU7EW3ME3TFVAVZANCNFSM4YXXP4MA>
> .
>

seveneightn9ne commented:
> Is there a way to test the automatic change of the material design theme, or is that manually changing the time on your device the only way?

The app only has 3 options: dark, light, or follow system setting. You can toggle between those options in the settings. What mode it uses when you have 'follow system setting' is controlled by the [react-native Appearance library](https://docs.expo.io/guides/color-schemes/), which gets its setting from the OS. You can change the OS system setting, and it should reflect back in the app.

> Is it possible to enable the wiki on github and give me access to contribute to that?

Done! Feel free to add anything you think is useful!

> ImagePage always shows black background, regardless of theme, suspect for
completeness this shouldn't be the case. What is your thought?

That's intentional - the background of that page was black before I added dark mode. I think it works well to not visually distract from the image.
